lib/msal-common/src/client/SilentFlowClient.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { BaseClient } from "./BaseClient";
+import { ClientConfiguration } from "../config/ClientConfiguration";
+import { SilentFlowRequest } from "../request/SilentFlowRequest";
+import { AuthenticationResult } from "../response/AuthenticationResult";
+import { CredentialType } from "../utils/Constants";
+import { CacheRecord } from "../unifiedCache/entities/CacheRecord";
+import { IdTokenEntity } from "../unifiedCache/entities/IdTokenEntity";
+import { CacheHelper } from "../unifiedCache/utils/CacheHelper";
+import { AccessTokenEntity } from "../unifiedCache/entities/AccessTokenEntity";
+import { RefreshTokenEntity } from "../unifiedCache/entities/RefreshTokenEntity";
+import { ScopeSet } from "../request/ScopeSet";
+import { IdToken } from "../account/IdToken";
+import { TimeUtils } from "../utils/TimeUtils";
+import { RefreshTokenRequest } from "../request/RefreshTokenRequest";
+import { RefreshTokenClient } from "./RefreshTokenClient";
+import { ClientAuthError } from "../error/ClientAuthError";
+import { CredentialFilter, CredentialCache } from "../unifiedCache/utils/CacheTypes";
+
+export class SilentFlowClient extends BaseClient {
+
+    constructor(configuration: ClientConfiguration) {
+        super(configuration);
+    }
+
+    /**
+     * Retrieves a token from cache if it is still valid, or uses the cached refresh token to renew
+     * the given token and returns the renewed token
+     * @param request
+     */
+    public async acquireToken(request: SilentFlowRequest): Promise<AuthenticationResult>{
+        let cacheRecord: CacheRecord;
+        let idTokenObj: IdToken;
+        const requestScopes = new ScopeSet(request.scopes || [], this.config.authOptions.clientId, true);
+
+        // We currently do not support silent flow for account === null use cases; This will be revisited for confidential flow usecases
+        if (request.account === null) {
+            throw ClientAuthError.createNoAccountInSilentRequestError();
+        } else {
+            // fetch account
+            const accountKey: string = CacheHelper.generateAccountCacheKey(request.account);
+            cacheRecord.account = this.unifiedCacheManager.getAccount(accountKey);
+
+            const homeAccountId = cacheRecord.account.homeAccountId;
+            const environment = cacheRecord.account.environment;
+
+            // fetch idToken, accessToken, refreshToken
+            cacheRecord.idToken = this.fetchIdToken(homeAccountId, environment, cacheRecord.account.realm);
+            idTokenObj = new IdToken(cacheRecord.idToken.secret, this.config.cryptoInterface);
+            cacheRecord.accessToken = this.fetchAccessToken(homeAccountId, environment, requestScopes, cacheRecord.account.realm);
+            cacheRecord.refreshToken = this.fetchRefreshToken(homeAccountId, environment);
```

Comment:
Adding a constructor to `CacheRecord` could make this code cleaner and easier to read/understand:

```js
const account = this.unifiedCacheManager.getAccount(accountKey);

const idToken = this.fetchIdToken(account.homeAccountId, account.environment, cacheRecord.account.realm);
const idTokenObj =  new IdToken(idToken.secret, this.config.cryptoInterface);

const accessToken = this.fetchAccessToken(account.homeAccountId, account.environment, requestScopes, account.realm);
const refreshToken = this.fetchRefreshToken(account.homeAccountId, account.environment);

const cacheRecord = new CacheRecord(idToken, accessToken, refreshToken);
```

Reply:
done, and I removed instances of CacheRecord from these files since we don't really need them.